### ISSUE TITLE
Wireframes: Advanced search

### ISSUE THREAD

**Issue by rlafuente** [beabee-communityrm/beabee-frontend#259]
We should have a quick call so that we can discuss the implementation details before moving to the spec, but here is our current vision for the advanced search interface.

[Open in PenPot](https://design.penpot.app/#/view/1bf90620-292f-11ed-b8ab-cf2300df1f4f?page-id=4e29b270-b03a-11ec-8735-b3cbc25196fd&section=interactions&index=0&share-id=4e21bda0-2a14-11ed-b8ab-cf2300df1f4f)

### Filter view after clicking "Advanced search"

![Advanced search](https://user-images.githubusercontent.com/159868/187966042-2de25b80-97e2-4d26-9d5d-6e6ebde134b6.png)

### Edit alternative 1 (and/or buttons)

![Advanced search: Editing](https://user-images.githubusercontent.com/159868/187966108-da7dc224-4cb4-4513-b017-ddad1f0b0bf1.png)

### Edit alternative 2 (match all/any)

![Advanced search: Editing (alt)](https://user-images.githubusercontent.com/159868/187966141-8663442f-7ed4-498c-8173-d4419ece4795.png)

**Comment by wpf500** [comment]
Looks simple enough :).

Firstly: we won't allow condition grouping or different AND/OR operators so edit alternative 2 (match all/any) is definitely the way to go here.

I'm curious how more "complex" filters will be edited, by which I mean filters that need more than one parameter. For example:
* Filtering "between" dates (i.e. needs two date fields)
* Filtering location fields (e.g. all contacts with x miles of a point)
* Filtering for roles (e.g. all contacts who were admins 2 years ago)

We don't need to design all of these filters now but we do need to think about how the UI can support complex filters that will presumably need more space than would be available on one line. This is where the current UI fails badly:

![image](https://user-images.githubusercontent.com/2084823/188861111-22e265e5-9db0-46a9-ba8c-67bb75ddebfa.png)

**Comment by aiscarvalho** [comment]
We started drafting more wireframes to address the possible lack of space and complex filters. Good points!
Here's a draft of the wireframes in a smaller screen. Drawing this also made us realise we're better off addressing space/layout in the spec, as we design the final UX elements that compose the advanced search :)

![image](https://user-images.githubusercontent.com/122730/189164767-a3a7d6b1-84ee-4ec2-bd7c-e21e5da7497a.png)

When we did the wireframes we didn't had in mind such complex filtering, especially things that would require conditional logic as the role example: first you filter by role and then are offered options such as setting a date (or date range) for the when the role was active.
It would be useful to map all the filters and subsequent conditionals that are planned for the MVP so we can design for them. For example, if role only had the sub condition of setting a date/range that's simpler than if there's multiple possible sub conditions.

To filter contacts within X miles of a point do we consider adding a map or input fields for the point address?

Maybe we discuss these details in a quick call?
And while that's still open we'll start working on the spec.

**Comment by aiscarvalho** [comment]
We started drafting more wireframes to address the possible lack of space and complex filters. Good points!
Here's a draft of the wireframes in a smaller screen. Drawing this also made us realise we're better off addressing space/layout in the spec, as we design the final UX elements that compose the advanced search :)

![image](https://user-images.githubusercontent.com/122730/189164767-a3a7d6b1-84ee-4ec2-bd7c-e21e5da7497a.png)

When we did the wireframes we didn't had in mind such complex filtering, especially things that would require conditional logic as the role example: first you filter by role and then are offered options such as setting a date (or date range) for the when the role was active.
It would be useful to map all the filters and subsequent conditionals that are planned for the MVP so we can design for them. For example, if role only had the sub condition of setting a date/range that's simpler than if there's multiple possible sub conditions.

To filter contacts within X miles of a point do we consider adding a map or input fields for the point address?

Maybe we discuss these details in a quick call?
And while that's still open we'll start working on the spec.
